### DISCUSSION TITLE
Cell proximity enrichment for features?

### DISCUSSION THREAD

**Discussion by lima1** [giotto-suite/Giotto#685] [Q&A]
Hi Giotto Team,

I'm interested in doing a cell proximity analysis asking whether cells expressing a certain gene are close to a certain cell type. I'm looking at Visium data and DWLS in general worked pretty well. I could create a fake DWLS with the cell types and genes of interest, but maybe I'm missing an existing solution?

On a semi-related question, the findICFSpot function, how do I generate the averaged expression matrix? Similar to the sign matrix for DWLS in single cell, but for all genes? Apologies if I missed an example in the documentation here.

Thanks a lot,
Markus

**Comment by RubD** [comment]
Hi @lima1 

I think `findICF` is more or less doing that exact analysis. It tries to identify genes that are up- or downregulated between specific sets of neighboring cell types (e.g. cell type A and B).

However, it's hard to interpret these results with Visium data where your spatial unit is actually a spot that covers anywhere between 5 and 50 cells. In other words, it's not easy to assess if the gene is expressed in the cell type of interest or in a neighboring cell.

Does that makes sense?

Ruben

**Reply by lima1** [reply]
Makes sense, thanks Ruben. I was hoping that findICFSpots at least gives you a good idea over many spots. So it's fine to use the average cell-type counts in single cell, normalized the same way as Visium, as averaged expression matrix, right?
